evm_loader/lib/src/signer.rs

```diff
@@ -0,0 +1,81 @@
+use std::sync::Arc;
+
+use solana_clap_utils::keypair::signer_from_path;
+use solana_sdk::{
+    pubkey::Pubkey,
+    signature::Signature,
+    signer::{Signer, SignerError},
+};
+
+use crate::{Config, NeonError};
+
+#[derive(Clone)]
+pub struct NeonSigner {
+    signer: Arc<dyn Signer>,
```

Comment:
I don't think that this implementation is safe because somewhere there are still might exist other references. If Signer's implementation is able to do interior mutability, for example, by using `RefCell`, there can be UB. Can we use `Box<dyn Signer>` here?

Reply:
The `Signer` implementations are all `Send` and `Sync`, so they do not have interior mutability. I need it to be an `Arc` because some places of the code require shared ownership because of async things.

Reply:
I just found it easier to do it this way.